analyzers/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -92,6 +92,25 @@ internal SonarAnalysisContext(AnalysisContext context, IEnumerable<DiagnosticDes
         public bool ShouldAnalyzeGenerated(Compilation c, AnalyzerOptions options) =>
             ShouldAnalyzeGenerated(context, c, options);
 
+        public bool IsTestProject(Compilation c, AnalyzerOptions options) =>
+            IsTestProject(context.TryGetValue, c, options);
+
+        public static bool IsTestProjectNotCached(Compilation c, AnalyzerOptions options)
+        {
+            var projectType = ProjectType.Unknown;
+            if (options.AdditionalFiles.FirstOrDefault(IsSonarProjectConfig) is { } sonarProjectConfigXml &&
+                sonarProjectConfigXml.GetText() is { } sourceText)
+            {
+                var projectConfigReader =  new ProjectConfigReader(sourceText, SonarProjectConfigFileName);
+                projectType = projectConfigReader.ProjectType;
+            }
+
+            return IsTest(projectType, c);
+        }
+
+        public static bool IsTestProject(CompilationAnalysisContext analysisContext) =>
```

Comment:
_Code orientation:_ Can you swap this method and the one above it? To have non-static `IsTestProject` followed by this static `IsTestProject` and then the static `IsTestProjectNoCache`?